fdbserver/OldTLogServer_6_0.actor.cpp

```diff
@@ -1228,21 +1285,173 @@ ACTOR Future<Void> tLogCommit(
 		return Void();
 	}
 
+	// while exec op is being committed, no new transactions will be admitted.
+	// This property is useful for snapshot kind of operations which wants to
+	// take a snap of the disk image at a particular version (no data from
+	// future version to be included)
+	if (logData->execOpCommitInProgress) {
+		wait(logData->execOpHold.getFuture());
+	}
+
+	state Version execVersion = invalidVersion;
+	state ExecCmdValueString execArg();
+	state TLogQueueEntryRef qe;
+	state StringRef execCmd;
+	state StringRef param2;
+	state vector<Tag> execTags;
+
 	if (logData->version.get() == req.prevVersion) {  // Not a duplicate (check relies on no waiting between here and self->version.set() below!)
 		if(req.debugID.present())
 			g_traceBatch.addEvent("CommitDebug", tlogDebugID.get().first(), "TLog.tLogCommit.Before");
 
-		//TraceEvent("TLogCommit", logData->logId).detail("Version", req.version);
-		commitMessages(self, logData, req.version, req.arena, req.messages);
-
-		logData->knownCommittedVersion = std::max(logData->knownCommittedVersion, req.knownCommittedVersion);
-
 		// Log the changes to the persistent queue, to be committed by commitQueue()
-		TLogQueueEntryRef qe;
 		qe.version = req.version;
 		qe.knownCommittedVersion = logData->knownCommittedVersion;
 		qe.messages = req.messages;
 		qe.id = logData->logId;
+
+		if (req.hasExecOp) {
+			// inspect the messages to find if there is an Exec type and print
```

Comment:
Can you move this into a function so that the size of tLogCommit stays reasonable?

Reply:
moved the second half of snap create to another function too.